airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_ads_insights_action_product_id.py

```diff
@@ -0,0 +1,489 @@
+    )
+
+
+@freezegun.freeze_time(NOW.isoformat())
+class TestFullRefresh(TestCase):
+
+    @staticmethod
+    def _read(config_: ConfigBuilder, expecting_exception: bool = False) -> EntrypointOutput:
+        return read_output(
+            config_builder=config_,
+            stream_name=_STREAM_NAME,
+            sync_mode=SyncMode.full_refresh,
+            expecting_exception=expecting_exception,
+        )
+
+    @HttpMocker()
+    def test_given_one_page_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(get_account_request().build(), get_account_response())
+        http_mocker.get(_update_api_throttle_limit_request().build(), _update_api_throttle_limit_response())
+        http_mocker.post(_job_start_request().build(), _job_start_response(_REPORT_RUN_ID))
+        http_mocker.post(_job_status_request(_REPORT_RUN_ID).build(), _job_status_response(_JOB_ID))
+        http_mocker.get(
+            _get_insights_request(_JOB_ID).build(),
+            _insights_response().with_record(_ads_insights_action_product_id_record()).build(),
+        )
+
+        output = self._read(config())
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_given_multiple_pages_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(get_account_request().build(), get_account_response())
+        http_mocker.get(_update_api_throttle_limit_request().build(), _update_api_throttle_limit_response())
+        http_mocker.post(_job_start_request().build(), _job_start_response(_REPORT_RUN_ID))
+        http_mocker.post(_job_status_request(_REPORT_RUN_ID).build(), _job_status_response(_JOB_ID))
+        http_mocker.get(
+            _get_insights_request(_JOB_ID).build(),
+            _insights_response().with_pagination().with_record(_ads_insights_action_product_id_record()).build(),
+        )
+        http_mocker.get(
+            _get_insights_request(_JOB_ID).with_pagination_parameter().build(),
+            _insights_response().with_record(_ads_insights_action_product_id_record()).with_record(
+                _ads_insights_action_product_id_record()
+            ).build(),
+        )
+
+        output = self._read(config())
+        assert len(output.records) == 3
+
+    @HttpMocker()
+    def test_given_multiple_days_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        start_date = NOW.subtract(days=1)
+        end_date = NOW
+        report_run_id_1 = "1571860060019500"
+        report_run_id_2 = "4571860060019599"
+        job_id_1 = "1049937379601600"
+        job_id_2 = "1049937379601699"
+
+        http_mocker.get(get_account_request().build(), get_account_response())
+        http_mocker.get(_update_api_throttle_limit_request().build(), _update_api_throttle_limit_response())
+        http_mocker.post(
+            _job_start_request(since=start_date, until=start_date).build(), _job_start_response(report_run_id_1)
```

Comment:
This is interesting to see that we send the same value for since and until. I understand that it is because the granularity of those dates are `days`. I think this is very good that it is explicit